src/bmffimage.cpp

```diff
@@ -452,6 +506,31 @@ uint64_t BmffImage::boxHandler(std::ostream& out /* = std::cout*/, Exiv2::PrintS
     case TAG_xml:
       parseXmp(buffer_size, io_->tell());
       break;
+    case TAG_brob: {
+      enforce(data.size() >= 4, Exiv2::ErrorCode::kerCorruptedMetadata);
+      uint32_t realType = data.read_uint32(0, endian_);
+      if (bTrace) {
+        out << "type: " << toAscii(realType);
+      }
+#ifdef EXV_HAVE_BROTLI
+      DataBuf arr;
+      brotliUncompress(data.c_data(4), data.size() - 4, arr);
+      if (realType == TAG_exif) {
+        uint32_t offset = arr.read_uint32(0, endian_) + 4;
+        enforce(offset + 4 < arr.size(), Exiv2::ErrorCode::kerCorruptedMetadata);
```

Comment:
The `offset + 4` could overflow here, allowing the `enforce` to be bypassed. I assume the purpose of this `enforce` is to prevent an integer overflow in `arr.size() - offset` below, so the `+4` looks unnecessary to me.
 
```suggestion
        enforce(offset < arr.size(), Exiv2::ErrorCode::kerCorruptedMetadata);
```

Reply:
The +4 is to account for the uint32 just read.

Reply:
In that case:

```suggestion
        enforce(Safe::add(offset, 4u) < arr.size(), Exiv2::ErrorCode::kerCorruptedMetadata);
```